File: backend/src/main/java/com/example/planesales/Controller/Request/ReviewRequest.java
package com.example.planesales.Controller.Request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

public record ReviewRequest(String reviewObject,
                            String objectId,
                            String reviewText,
                            byte[] reviewPicture,
                            @Pattern(regexp = "^[1-5]$")
                            @NotBlank(message = "Rating can not be blank")
                            int reviewRating) {
}

File: Frontend/planesales-react/src/schema/ReviewRequest.js
class ReviewRequest {
    constructor(reviewObject = null, objectId = null, reviewText = null, reviewPicture = null, reviewRating = null) {
        this.reviewObject = reviewObject;
        this.objectId = objectId;
        this.reviewText = reviewText;
        this.reviewPicture = reviewPicture;
        this.reviewRating = reviewRating;
    }
}

export default ReviewRequest;
